README.md

```diff
@@ -1,3 +1,55 @@
+
+### Authentication
+
+Users register and log in to the system through 'magic-links', these are one-time links that are sent into user's email inbox.  The link contains an authCode, that can be exchanged for a secure token used for future requests.  The link is one-time, as it expires on first usage.  To login again, a new magic link, containing a new auth code, must be requested.  As such, the system does not contain any passwords, and can only be accessed via the users email inbox.
+
+### Data capture
+
+The application asks users to submit both demographic (date of birth & sex) & location (full address) information, to enable demographic segmentation of results.  Address information may be used to verify a petients identity, and at later stages to track outbreaks via test results.  Data is only communicated over secure SSL connections.
+
+### Data access & sharing results
+
+The system is secured such that patients control their own data. A patient may issue a unique, short-lived, sharing code that enables another user to securely view and verify those results. Sharing codes have a life span of only 60 seconds to prevent their leakage and to prevent any brute force attempt to crack codes.
+
+A sharing codes will typically be converted into a QR code which can be captured by another user. On capture, the capturer is granted an 'access pass' giving them permissions to view the sharers test result data (currently for one hour).  There is no other way to lookup or view a patients data, ensuring the patient controls who can access their data.
+
+If the capturing user has sufficient permission (e.g. they are a registered doctor), they may add new results on behalf of a patient.
+
+### Roles, permissions & test confidence
+
+The platform contains a flexible system for managing roles & permissions.  These enable priviledged users to perform more tasks (like submitting a test result for another user).  They also influence the confidence level in a test result.  
+
+If a test is administered and processed by a medical professional with a recognised role, we can have high confidence that the tests was perfomed by the correct patient, and that the result was entered faithfully.  If a patient performs their own test & self submits data, we cannot be sure that someone else did not take the test for them, nor that they submitted the correct data. The system records test confidence based on the roles of the users submitting the data.
+
+### Data analysis
+
+Data analysis is performed on anonymised data sets, and by authorised personel only.  Test subject identifiers are anonymised with a one-way hash, allowing the analysis of how test results for a single patient evolve over time, but without exposing any link back to the original patient.  Hashes are 'salted' such that it is not possible to deanonymise test result data if you were to discover a user's identifier and which hash algorithm is in use.  
```

Comment:
This para needs work - the code is open source, don't think "discovering the algorithm" is relevant. I would emphasize more what we talked about on slack - not the hashing but rather the random salt per export.